pageserver/src/lib.rs

```diff
@@ -254,3 +267,36 @@ mod backoff_defaults_tests {
         );
     }
 }
+
+#[cfg(test)]
+mod timed_tests {
+    use super::timed;
+    use std::time::Duration;
+
+    #[tokio::test]
+    async fn test_timed() {
+        // A future that completes on time should have its result returned
+        let r1 = timed(
+            async move {
+                tokio::time::sleep(Duration::from_millis(10)).await;
```

Comment:
tokio shouldn't be sleeping here at all, so I think we could just remove the sleep. It's timer facilities should be coarse grained to 100ms.

Reply:
It's meant to be 1ms resolution according to here (I think historic versions were much coarser):
https://docs.rs/tokio/latest/src/tokio/runtime/time/mod.rs.html#43

Reply:
Ah interesting. Well, I don't think the test needs it either way.

Reply:
If you wanted to ensure that the future is in fact a generator, one could `tokio::task::yield_now().await` instead of the sleep? Then the intention would be more clear.

Reply:
Ah nevermind, I failed to see this is above and the other one is below the `warn_at`. Okay, maybe I can live with these. Annoyed though that we cannot assert the log messages...